cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -366,6 +370,17 @@ public void setVersion(VersionIdentifier version, boolean hint) {
     }
   }
 
+  private void checkVersionExistence(String tool, String edition, VersionIdentifier version) {
+
+    try {
+      UrlMetadata metadata = this.context.getUrls();
+      metadata.getVersionFolder(tool, edition, version);
+    } catch (IllegalArgumentException e) {
+      throw new CliException(
+          "Version " + version + " for tool " + tool + " does not exist in edition " + edition + ".");
+    }
```

Comment:
The method `getVersion` already throws a `CliException` if a version pattern is given but no matching version could be found.
IMHO it would make sense to change `getVersionFolder` to throw `CliException` instead of `IllegalArgumentException` and avoid this catch block. And be aware that such "exception mapping" is typically kind of unsafe as some other developer might add another `if` check also throwing `IllegalArgumentException` for a different purpose and then we would cause a lot of confusion with this. Also you are not passing the catched exception `e` to the new exception.

Reply:
Perfect, makes totally sense. Then the check could be done directly in one line in `setVersion `so having a function `checkVersionExistence` doesent make sense anymore.